lib/main.js

```diff
@@ -95,15 +97,15 @@ const Hydrogen = {
         const editorSubscriptions = new CompositeDisposable();
         editorSubscriptions.add(
           editor.onDidChangeGrammar(() => {
-            store.setGrammar(editor);
+            store.updateEditorAndGrammar(editor);
           })
         );
 
         if (isMultilanguageGrammar(editor.getGrammar())) {
           editorSubscriptions.add(
             editor.onDidChangeCursorPosition(
               _.debounce(() => {
-                store.setGrammar(editor);
+                store.updateEditorAndGrammar(editor);
```

Comment:
Why did you merge `setGrammar` and `updateEditor` into `updateEditorAndGrammar`?

In this case the editor doesn't change, yet we update it in the store. Theoretically this should be fine but I'm not sure if it can cause mobx to trigger a rerender somewhere.